pydantic/_internal/_generate_schema.py

```diff
@@ -568,7 +579,9 @@ def _namedtuple_schema(self, namedtuple_cls: Any) -> core_schema.CallSchema:
         """
         Generate schema for a NamedTuple.
         """
-        annotations: dict[str, Any] = _typing_extra.get_type_hints(namedtuple_cls, include_extras=True)
+        annotations: dict[str, Any] = get_type_hints_infer_globalns(
```

Comment:
It seems to me the same infer-globalns logic deserved to apply here as well. (I found this by doing find-everywhere search for `get_type_hints`.)